lib/pure/httpclient.nim

```diff
@@ -1159,6 +1164,12 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url: string,
   # Helper that actually makes the request. Does not handle redirects.
   let requestUrl = parseUri(url)
 
+  when client is AsyncHttpClient:
+    if (not client.pending.isNil) and (not client.pending.finished):
```

Comment:
``and (not client.pending.finished)`` isn't necessary. ``await`` will be a no-op if `pending` is finished.